### PR TITLE
Improve wording of Changing the Colors » System preference

### PR DESCRIPTION
**Pull request by posita**
Motivated by #2727.

### CODE REVIEW

File: docs/setup/changing-the-colors.md [hunk 1]
```diff
@@ -208,12 +208,12 @@ The following fields must be set for each toggle:
   [9]: #accent-color
   [10]: https://github.com/squidfunk/mkdocs-material/tree/master/material/.icons
 
-### System preference
+### Inferring the initial setting from system preference
```
Comment (squidfunk) [inline docs/setup/changing-the-colors.md:211]: We shouldn't change the headline title, because we'd lose all anchor links. Please change it back.

File: docs/setup/changing-the-colors.md [hunk 2]
```diff
@@ -208,12 +208,12 @@ The following fields must be set for each toggle:
...
 
 [:octicons-file-code-24: Source][6] · :octicons-milestone-24: Default: _none_
 
-In order to automatically set the color palette to the user's system preference,
-a media query can be set as part of the `media` field next to the toggle
+To match a color palette to the user's system preference upon the user's first
```
Comment (squidfunk) [inline docs/setup/changing-the-colors.md:215]: I find "to match ..." less intuitive.

**Review by squidfunk — CHANGES_REQUESTED** [review]

File: docs/setup/changing-the-colors.md (hunk 2 above)
Comment (posita) [reply docs/setup/changing-the-colors.md:215]: What would you suggest as an alternative?

Comment (posita) [reply docs/setup/changing-the-colors.md:215]: I kept the "To match …" for now, but word-smithed the remainder a bit. Hopefully that helps?

Comment (squidfunk) [reply docs/setup/changing-the-colors.md:215]: No, not really.

File: docs/setup/changing-the-colors.md [hunk 3]
```diff
@@ -212,28 +212,28 @@ The following fields must be set for each toggle:
...
-a media query can be set as part of the `media` field next to the toggle
-definition in `mkdocs.yml`:
+To match a color palette to the user's system preference upon the user's first
+visit, provide a media query via the `media` field next to the `scheme` definition
+in `mkdocs.yml`:
 
 ``` yaml hl_lines="3 8"
 theme:
   palette:
-    - media: "(prefers-color-scheme: light)"
```
Comment (squidfunk) [inline docs/setup/changing-the-colors.md:222]: I don't know whether I've not seen this in the first review, because you force-pushed (please don't do that, as we loose track of changes). 

Why did you change the order? First of all, now the wrong lines are highlighted, and second of all I prefer `media` to be at the top, because it's more clear that the following settings are tied to it.

**Review by squidfunk — CHANGES_REQUESTED** [review]

File: docs/setup/changing-the-colors.md (hunk 3 above)
Comment (posita) [reply docs/setup/changing-the-colors.md:222]: Yeah, this was new, because it dawned on me after discussing "match[ing]" above that the ordering seemed…_off_.

Thanks for reminding me about the highlighting! Should be fixed now.

My suggestion is to put `scheme` first, because it is the primary thing everything else is anchored. Both `media` and `toggle` are subordinate to `scheme` no? (Meaning neither makes sense without `scheme` being present, correct?)

File: docs/setup/changing-the-colors.md (hunk 2 above)
Comment (posita) [reply docs/setup/changing-the-colors.md:215]: If my original defense was unpersuasive, what about this?

> To ~match~ _set_ a color palette _according_ to the user's system preference upon the user's first …

FWIW, I really think my original proposal is not only more concise, but more plainly and directly describes what's going on. "To set … according to …" means "match" in this context. Matching is fundamentally what's being done, no?

File: docs/setup/changing-the-colors.md (hunk 3 above)
Comment (squidfunk) [reply docs/setup/changing-the-colors.md:222]: You can set any of the primary, accent or scheme values regardless of each other, so there’s no parent relationship with the scheme.

The current order is more intuitive because it read like “for light mode I set primary to ...”

Comment (posita) [reply docs/setup/changing-the-colors.md]: Sorry for the confusion. I have reverted this.

For what it's worth, I was suggesting they were semantically dependent (at least given what was present in the example), not syntactically. I realize that they are peers in the YAML schema. I only meant to highlight that `media` and `toggle` are effectively meaningless without at least one of `primary`, `accent`, or `scheme` (and only `scheme` was present in the example). Consider, for example, the following configuration:

```yaml
theme:
  palette:
    - media: "(prefers-color-scheme: light)"
      toggle:
        icon: material/toggle-switch-off-outline
        name: Switch to some mode
    - media: "(prefers-color-scheme: dark)"
      toggle:
        icon: material/toggle-switch
        name: Switch to some other mode
```

In any event, I backed this out per your request.

File: docs/setup/changing-the-colors.md (hunk 2 above)
Comment (posita) [reply docs/setup/changing-the-colors.md:215]: Any suggestions then?

If my original defense was unpersuasive, what about this?

> To ~match~ _set_ a color palette _according_ to the user's system preference upon the user's first …

FWIW, I really think my original proposal is not only more concise, but more plainly and directly describes what's going on. "To set … according to …" means "match" in this context. Matching is fundamentally what's being done, no?

**Comment by squidfunk** [comment]
Honestly, I find the original paragraphs (both) better as I find them easier to understand than the current draft.

**Comment by posita** [comment]
It's your baby. 😊 I'll withdraw the PR. Thanks for the consideration, though!

Not to beat a dead horse, but I think this passage as written is probably accessible to someone who already understands the subject matter (such as yourself). Consider, though, that human beings often have a difficult time empathizing with others who lack similar context or expertise.

The passage—particularly the phrase, "… automatically set the color palette to the user's system preference …"—left me with a fundamental misunderstanding of even the desired behavior. So much so that I went through the effort to file #2727 thinking the behavior was somehow incorrect, and then followed up trying to addressing the source of my confusion in this PR. That being said, I'm a sample size of one, and not so arrogant to assume my experience is representative, either. I'm open to the possibility that I am an outlier.

No worries if you don't think my edits are helpful. I appreciate the time, attention, and consideration. Please consider revisiting this topic if similar confusion arises from another source.

**Comment by squidfunk** [comment]
Thanks for your understanding! IIRC you're the first one being tripped up by the wording, so I think we're not really solving a problem here. However, we'll definitely revisit if the need arises.
